rdrf/rdrf/admin.py

```diff
@@ -178,6 +183,87 @@ def get_readonly_fields(self, request, obj=None):
         return () if obj is None else ("code",)
 
 
+class ActivationKeyExpirationListFilter(admin.SimpleListFilter):
+    title = _('activation key expired')
+    parameter_name = 'activation_key_expired'
+
+    def lookups(self, request, model_admin):
+        return [
+            ('True', _('Yes')),
+            ('False', _('No')),
+        ]
+
+    def queryset(self, request, queryset):
+        expired_profiles = [profile.id for profile in queryset if profile.activation_key_expired()]
+
+        if self.value() == 'True':
+            return queryset.filter(id__in=expired_profiles)
+        elif self.value() == 'False':
+            return queryset.exclude(id__in=expired_profiles)
+        else:
+            return queryset
+
+
+def resend_activation_mail(profile, site, request=None):
+    if not hasattr(profile.user, 'registrationprofile') or profile.activated:
+        return False
+
+    profile.create_new_activation_key()
+    profile.send_activation_email(site, request)
+
+    return True
+
+
+class CustomRegistrationProfileAdmin(RegistrationAdmin):
+    list_display = ('user', 'custom_activation_key_expired', 'activated')
+    list_filter = ['activated', ActivationKeyExpirationListFilter]
+
+    @admin.display(description="Activation key expired")
+    def custom_activation_key_expired(self, obj):
+        max_expiry_days = datetime.timedelta(days=settings.ACCOUNT_ACTIVATION_DAYS)
+        activation_date = obj.user.date_activated
+        expiration_date = (obj.user.date_joined if activation_date is None else activation_date) + max_expiry_days
```

Comment:
I think you can do:
`(activation_date or obj.user.date_joined) + max_expiry_days` to achieve the same logic